crates/red_knot_python_semantic/resources/mdtest/binary/integers.md

```diff
@@ -10,9 +10,10 @@ reveal_type(-3 // 3)  # revealed: Literal[-1]
 reveal_type(-3 / 3)  # revealed: float
 reveal_type(5 % 3)  # revealed: Literal[2]
 
-# TODO: We don't currently verify that the actual parameter to int.__add__ matches the declared
-# formal parameter type.
-reveal_type(2 + "f")  # revealed: int
+# TODO: This should emit an unsupported-operator error but we don't currently
+#  verify that the actual parameter to `int.__add__` matches the declared
+#  formal parameter type.
```

Comment:
I think this comment is now wrong, not in terms of what should happen (yes, there should be an unsupported-operator error), but the why. What is happening here after this PR is that we are falling back to `str.__radd__` (because we have the wrong argument type for `int.__add__` -- we _do_ now "verify that the actual parameter to `int.__add__` matches the declared formal parameter type"), but `str.__radd__` is overloaded and we don't understand that, so we fail to detect that this operation is also invalid for `str.__radd__`.

Reply:
I'll put up a quick PR for this; it's minor but I don't want to waste the time I spent understanding what's happening.

Reply:
https://github.com/astral-sh/ruff/pull/16242